tests/js/browser-binder-tests.js

```diff
@@ -0,0 +1,639 @@
+// Test "binder" components using `gpii-test-browser`.
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+require("gpii-test-browser");
+gpii.tests.browser.loadTestingSupport();
+
+var path    = require("path");
+var url     = require("url");
+var fileUrl = url.resolve("file://", path.resolve(__dirname, "../static/tests-binder.html"));
+
+fluid.registerNamespace("gpii.binder.tests");
+
+// TODO:  If we find ourselves making model changes often, move this to `gpii-test-browser` and polish it up for wider use.
+gpii.binder.tests.applyChange = function (componentPath, modelPath, newValue) {
+    var component = fluid.getGlobalValue(componentPath);
+    component.applier.change(modelPath, newValue);
+};
+
+
+fluid.defaults("gpii.binder.tests.caseHolder", {
```

Comment:
OK, I took a minute to read up on `testem` and see how it's used in `infusion`, which uses `QUnit` and [`jquery.simulate`](https://github.com/jquery/jquery-simulate).  I now have a better response.  For pure client-side unit tests, `testem` and `QUnit` are fine.  The key here is how we simulate browser events.

Infusion uses `jquery.simulate`, which still exists, but its maintainers are no longer updating it.  They are moving towards the [WebDriver API](https://www.w3.org/TR/webdriver/) and [Intern](https://theintern.github.io/).

Nightmare uses its own similar (and similarly flawed) javascript trickery to simulate events, but my hope has been that eventually they will integrate more tightly with the Atom Electron browser and be able to trigger browser events directly.  For now, though, there's not much difference, and Nightmare will never be able to test compatibility with a range of browsers.

So, I agree that we need something like `testem` or `Intern` for this package, as the current tests only test one Chrome variant.  However, I am reluctant to put the time into to converting to `jquery.simulate` without at least looking around briefly for other options.  We can discuss in our next chat or (probably better) take it to the next architecture meeting to discuss there.